compiler/rustc_session/src/config.rs

```diff
@@ -1159,6 +1159,31 @@ impl CrateCheckConfig {
         self.fill_well_known_names();
         self.fill_well_known_values();
     }
+
+    pub fn warn_actual(&self, opts: &Options, cfg: &CrateConfig) {
+        for &(name, value) in cfg {
+            if let Some(names_valid) = &self.names_valid {
+                if !names_valid.contains(&name) {
+                    early_warn(
```

Comment:
Is there a better way to emit a warning than using `early_warn` (which I believe cannot be enable/disable/promoted to error/...) ?

Reply:
Command line options are not tied to AST nodes, so I guess crate root is the only reasonable place to put attributes that can allow/deny/etc this warning.
Lint with a node can be added with `buffer_lint` as soon as `ParseSess` is available.

Reply:
I guess, the simplest solution would be to check this during the main early lint pass.

Reply:
I used the previous lint and added to it a early lint pass implementation. I ended up not using `parse_sess` but directly using the methods on `EarlyContext`. I didn't even needed to use `CRATE_NODE_ID`.